src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,259 @@ void DOS_Shell::CMD_VOL(char* args)
 	const auto trimmed_label = trim(label);
 	WriteOut(MSG_Get("SHELL_CMD_VOL_OUTPUT"), drive, trimmed_label, high, low);
 }
+
+void DOS_Shell::CMD_MOVE(char* args)
+{
+	HELP("MOVE");
+
+	// TODO: Add support for these flags along with the COPYCMD environment
+	// variable (overwrite prompts) Also currently ignored by CMD_COPY
+	(void)ScanCMDBool(args, "Y");
+	(void)ScanCMDBool(args, "-Y");
+
+	char* rem = ScanCMDRemain(args);
+	if (rem) {
+		WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"), rem);
+		return;
+	}
+
+	bool parsing          = true;
+	bool quote            = false;
+	bool stop_next_letter = false;
+	bool comma            = false;
+	std::string word;
+	std::vector<std::string> raw_sources;
+
+	// Parse sources (seperated by commas)
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			quote = !quote;
+			args++;
+			break;
+		case ' ':
+			if (quote) {
+				word.push_back(*args++);
+			} else {
+				if (!word.empty()) {
+					raw_sources.emplace_back(std::move(word));
+					word             = {};
+					stop_next_letter = true;
+					comma            = false;
+				}
+				args++;
+			}
+			break;
+		case ',':
+			stop_next_letter = true;
+			comma            = true;
+			if (!word.empty()) {
+				raw_sources.emplace_back(std::move(word));
+				word = {};
+			}
+			args++;
+			break;
+		case 0: parsing = false; break;
+		default:
+			if (stop_next_letter) {
+				parsing          = comma;
+				comma            = false;
+				stop_next_letter = false;
+			} else {
+				word.push_back(*args++);
+			}
+		}
+	}
+
+	if (raw_sources.empty()) {
+		WriteOut(MSG_Get("SHELL_MISSING_PARAMETER"));
+		return;
+	}
+
+	std::string raw_destination;
+	parsing = true;
+
+	// Parse detination
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			quote = !quote;
+			args++;
+			break;
+		case ' ':
+			if (quote) {
+				raw_destination.push_back(*args++);
+			} else {
+				args++;
+				parsing = false;
+			}
+			break;
+		case '\0': parsing = false; break;
+		default: raw_destination.push_back(*args++);
+		}
+	}
```

Comment:
Same comment as above about the parsing; at the very least this could go into a separate function to split this big function up a little bit into more logical chunks.